modules/sor/sorV2/sorV2.service.ts

```diff
@@ -0,0 +1,332 @@
+        const tIn = await this.getToken(tokenIn as Address, chainId);
+        const tOut = await this.getToken(tokenOut as Address, chainId);
+        const swapKind = this.mapSwapType(swapType);
+        try {
+            // Constructing a Swap mutates the pools so I used cloneDeep
+            const swap = await sorGetSwapsWithPools(
+                        tIn,
+                        tOut,
+                        swapKind,
+                        swapAmount,
+                        cloneDeep(poolsFromDb),
+                        // swapOptions, // I don't think we need specific swapOptions for this?
+                    );
+            return new SwapResultV2(swap);
+        } catch (err) {
+            console.log(`sorV2 Service Error`, err);
+            return new SwapResultV2(null);
+        }
+    };
+
+    /**
+     * Gets a b-sdk Token based off tokenAddr.
+     * @param tokenAddr 
+     * @param chainId 
+     * @returns 
+     */
+    private async getToken(tokenAddr: Address, chainId: ChainId): Promise<Token> {
+        const tokens = await tokenService.getTokens();
+        const prismaToken = this.getPrismaToken(tokenAddr, tokens);
+        return new Token(
+            chainId,
+            tokenAddr,
+            prismaToken.decimals,
+            prismaToken.symbol,
+        );
+    }
+
+    private getPrismaToken(tokenAddress: string, tokens: PrismaToken[]): PrismaToken {
+        tokenAddress = tokenAddress.toLowerCase();
+        const match = tokens.find((token) => token.address === tokenAddress);
+
+        if (!match) {
+            throw new Error('Unknown token: ' + tokenAddress);
+        }
+        return match;
+    }
+
+    private mapSwapType(swapType: GqlSorSwapType): SwapKind {
+        return swapType === "EXACT_IN" ? SwapKind.GivenIn : SwapKind.GivenOut;
+    }
+
+    private async getBasePools(): Promise<BasePool[]> {
+        let basePools: BasePool[] | null = this.cache.get(`${ALL_BASEPOOLS_CACHE_KEY}:${networkContext.chainId}`);
+        if (!basePools) {
+            basePools = await this.getBasePoolsFromDb();
+            this.cache.put(`${ALL_BASEPOOLS_CACHE_KEY}:${networkContext.chainId}`, basePools, 5 * 60 * 1000);
+        }
+        return basePools;
+    }
+
+    /**
+     * Fetch pools from Prisma and map to b-sdk BasePool.
+     * @returns 
+     */
+    private async getBasePoolsFromDb(): Promise<BasePool[]> {
+        const pools = await prisma.prismaPool.findMany({
+            where: { 
+                chain: networkContext.chain,
+                dynamicData: {
+                    totalSharesNum: {
+                        gt: 0.000000000001,
+                    },
+                    swapEnabled: true,
+                },
+                NOT: {
+                    id: {
+                        in: networkContext.data.sor[env.DEPLOYMENT_ENV as DeploymentEnv].poolIdsToExclude,
+                    }
+                }
```

Comment:
this could be written as                 
id: {
        notIn: networkContext.data.sor[env.DEPLOYMENT_ENV as DeploymentEnv].poolIdsToExclude,
    },